### PR TITLE
*: support gbk charset

### PR DESCRIPTION
**Pull request by zimulala**
* Add gbk charset

* close https://github.com/pingcap/mysql-tester/issues/25. 
  * `MySQLWarnings` is removed in go-sql-driver/mysql(v1.6.0). So here remove it too.
  * `strict mode`  is removed too in https://github.com/go-sql-driver/mysql/pull/676.

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @morgo @tangenta @xiongjiwei

File: src/main.go [hunk 1]
```diff
@@ -155,7 +167,7 @@ func isTiDB(db *sql.DB) bool {
 }
 
 func (t *tester) addConnection(connName, hostName, userName, password, db string) {
-	mdb, err := OpenDBWithRetry("mysql", userName+":"+password+"@tcp("+hostName+":"+port+")/"+db+"?strict=true&time_zone=%27Asia%2FShanghai%27"+params)
+	mdb, err := OpenDBWithRetry("mysql", userName+":"+password+"@tcp("+hostName+":"+port+")/"+db+"?time_zone=%27Asia%2FShanghai%27"+params)
```
Comment (xiongjiwei) [inline src/main.go:170]: what is the `strict` mean, why we need remove it

Comment (dveeden) [reply src/main.go:170]: The strict mode is an option in the driver that changes warnings to errors. However in newer versions of the driver this has been removed.

https://github.com/go-sql-driver/mysql/wiki/strict-mode

**Review by xiongjiwei — APPROVED** [review]

**Comment by zimulala** [comment]
PTAL @morgo @tangenta

File: src/main.go [hunk 2]
```diff
@@ -800,6 +796,7 @@ func (t *tester) executeStmt(query string) error {
 			return errors.Trace(err)
 		}
 	}
+
```
Comment (bb7133) [inline src/main.go:799]: Please remove this blank line.

**Review by dveeden — APPROVED** [review]
I don't know what exact issue the GBK part fixes, so I can't fully review that part. However updating the dependencies looks ok and I don't see any issues with the GBK part of this.

File: src/main.go [hunk 3]
```diff
@@ -57,6 +57,18 @@ func init() {
...
+		Name:             "gbk",
+		DefaultCollation: "gbk_bin",
+		Collations:       map[string]*charset.Collation{},
+	}
+	charset.AddCharset(c)
+	for _, coll := range charset.GetCollations() {
+		if strings.EqualFold(coll.CharsetName, c.Name) {
+			charset.AddCollation(coll)
+		}
+	}
```
Comment (zimulala) [inline src/main.go:71]: GBK related issues are dealt with here. @dveeden

**Review by bb7133 — APPROVED** [review]

**Pull request merged**